build/psalm/OcpSinceChecker.php

```diff
@@ -32,11 +43,58 @@ public static function afterClassLikeVisit(AfterClassLikeVisitEvent $event): voi
 			}
 
 			if ($stmt instanceof EnumCase) {
-				self::checkStatementComment($stmt, $statementsSource, 'enum');
+				if ($isTesting) {
+					self::checkStatementAttributes($classLike, $statementsSource);
+				} else {
+					self::checkStatementComment($stmt, $statementsSource, 'enum');
+				}
 			}
 		}
 	}
 
+	private static function checkStatementAttributes(ClassLike $stmt, FileSource $statementsSource): void {
+		$hasAppFrameworkAttribute = false;
+		$mustBeConsumable = false;
+		$isConsumable = false;
+		foreach ($stmt->attrGroups as $attrGroup) {
+			foreach ($attrGroup->attrs as $attr) {
+				if (in_array($attr->name->getLast(), [
```

Comment:
```suggestion
				if (in_array($attr->name->namespacedName, [
```
Can you check if this is set? Then the full name will be used which would be safer

Reply:
I tried all my best before:
```
Undefined property: PhpParser\Node\Attribute::$namespacedName
```


Throwable had a `resolvedName` because it "conflicts" without a namespace, but all others did not have anything "full":
```
object(PhpParser\Node\Attribute)#24860 (3) {
  ["attributes":protected]=>
  array(3) {
    ["startLine"]=>
    int(13)
    ["startFilePos"]=>
    int(286)
    ["endFilePos"]=>
    int(314)
  }
  ["name"]=>
  object(PhpParser\Node\Name)#24864 (2) {
    ["attributes":protected]=>
    array(3) {
      ["startLine"]=>
      int(13)
      ["startFilePos"]=>
      int(286)
      ["endFilePos"]=>
      int(298)
    }
    ["parts"]=>
    array(1) {
      [0]=>
      string(13) "Implementable"
    }
  }
  ["args"]=>
  array(1) {
    [0]=>
    object(PhpParser\Node\Arg)#24861 (5) {
      ["attributes":protected]=>
      array(3) {
        ["startLine"]=>
        int(13)
        ["startFilePos"]=>
        int(300)
        ["endFilePos"]=>
        int(313)
      }
      ["name"]=>
      object(PhpParser\Node\Identifier)#24862 (2) {
        ["attributes":protected]=>
        array(3) {
          ["startLine"]=>
          int(13)
          ["startFilePos"]=>
          int(300)
          ["endFilePos"]=>
          int(304)
        }
        ["name"]=>
        string(5) "since"
      }
      ["value"]=>
      object(PhpParser\Node\Scalar\String_)#24863 (2) {
        ["attributes":protected]=>
        array(5) {
          ["startLine"]=>
          int(13)
          ["startFilePos"]=>
          int(307)
          ["endFilePos"]=>
          int(313)
          ["kind"]=>
          int(1)
          ["rawValue"]=>
          string(7) "'9.0.0'"
        }
        ["value"]=>
        string(5) "9.0.0"
      }
      ["byRef"]=>
      bool(false)
      ["unpack"]=>
      bool(false)
    }
  }
}
object(PhpParser\Node\Attribute)#58040 (3) {
  ["attributes":protected]=>
  array(3) {
    ["startLine"]=>
    int(14)
    ["startFilePos"]=>
    int(236)
    ["endFilePos"]=>
    int(261)
  }
  ["name"]=>
  object(PhpParser\Node\Name)#58043 (2) {
    ["attributes":protected]=>
    array(4) {
      ["startLine"]=>
      int(14)
      ["startFilePos"]=>
      int(236)
      ["endFilePos"]=>
      int(244)
      ["resolvedName"]=>
      string(36) "OCP\AppFramework\Attribute\Throwable"
    }
    ["parts"]=>
    array(1) {
      [0]=>
      string(9) "Throwable"
    }
  }
  ["args"]=>
  array(1) {
    [0]=>
    object(PhpParser\Node\Arg)#58042 (5) {
      ["attributes":protected]=>
      array(3) {
        ["startLine"]=>
        int(14)
        ["startFilePos"]=>
        int(246)
        ["endFilePos"]=>
        int(260)
      }
      ["name"]=>
      object(PhpParser\Node\Identifier)#58041 (2) {
        ["attributes":protected]=>
        array(3) {
          ["startLine"]=>
          int(14)
          ["startFilePos"]=>
          int(246)
          ["endFilePos"]=>
          int(250)
        }
        ["name"]=>
        string(5) "since"
      }
      ["value"]=>
      object(PhpParser\Node\Scalar\String_)#58044 (2) {
        ["attributes":protected]=>
        array(5) {
          ["startLine"]=>
          int(14)
          ["startFilePos"]=>
          int(253)
          ["endFilePos"]=>
          int(260)
          ["kind"]=>
          int(1)
          ["rawValue"]=>
          string(8) "'30.0.0'"
        }
        ["value"]=>
        string(6) "30.0.0"
      }
      ["byRef"]=>
      bool(false)
      ["unpack"]=>
      bool(false)
    }
  }
}
```

Reply:
PS: Sounds familiar to you I guess? https://github.com/nextcloud/openapi-extractor/blob/main/src/ControllerMethod.php#L341